src/utils/nameUtils.js

```diff
@@ -0,0 +1,9 @@
+export function deriveIndividualName(
+  individualData,
+  context = 'defaultName',
+  fallback = undefined,
+) {
+  const names = individualData?.names || [];
+  const nameObject = names.find(n => n?.context === context) || {};
+  return nameObject?.value || fallback;
```

Comment:
What is the scenario where `nameObject` is nullish?

Reply:
since the individual data comes back from the API we have to assume it could have pretty much anything wrong with it. for example if we received 

```
{
  ...individualProperties...
  name: [{ name1... }, undefined, { name2... }
}
```

we will hopefully render something janky around the weird value rather than fully crashing the page.

Reply:
What you currently have in line 7 (`const nameObject = names.find(n => n?.context === context) || {};`) should be handling that though. Either `names.find` will find the correct name object, in which case `nameObject` is an object with at least a context field, or `names.find` will return `undefined`, in which case your or operator sets `nameObject` to an empty object. I'm not seeing a case when `nameObject` is not an object.

Reply:
fair enough. do you think `?.` hurts anything?